sklearn/metrics/_classification.py

```diff
@@ -3589,8 +3591,17 @@ def brier_score_loss(
     ... )
     0.146...
     """
+    xp, _, device = get_namespace_and_device(
+        y_true,
+        y_proba,
+        sample_weight,
+    )
     y_proba = check_array(
-        y_proba, ensure_2d=False, dtype=[np.float64, np.float32, np.float16]
+        y_proba,
+        ensure_2d=False,
+        dtype=tuple(
+            xp.__array_namespace_info__().dtypes(kind="real floating").values()
```

Comment:
Do we really need this change aside from just replacing np with xp in the floatdata types? Is there some other float dtype that we want to support?

Reply:
I did this since some libraries like PyTorch MPS don't support xp.float32.
(Also float16 is not in the array API standard. Should we make a special exception for np.float16?)

Maybe it would be good to put this in a helper in the array API utils module?